etna/transforms/math/statistics.py

```diff
@@ -59,6 +60,10 @@ def __init__(
     def fit(self, ts: TSDataset) -> "WindowStatisticsTransform":
         """Fit the transform."""
         self.in_column_regressor = self.in_column in ts.regressors
+        if not self.in_column_regressor:
```

Comment:
I'm not sure if we should give this warning. 
A fairly valid approach is to apply `MeanTransform` to target and then apply `LagTransform` to the result. It will be strange to force someone to change the order if everything is fine.

However, with new warning we could draw users' attention to the issue. Failed tests are also bothering me, some of them could be problematic and probably should be fixed. 

- Keep this warning for a while (we will remove it a little bit later, see second step).
- Fix only the tests where we create a pipeline without lag applied to window transform.
- Ask for re-review (after that we probably remove this warning).

Reply:
We have no tests with pipeline

Reply:
Ok, we could remove this warning and make other suggested changes.